File: app/src/main/java/com/slimshady/bottomsheetmodule/BaseBottomSheet.java
package com.slimshady.bottomsheetmodule;



import static com.slimshady.bottomsheetmodule.util.ScreenUtils.getWindowHeight;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;

import androidx.viewbinding.ViewBinding;

import com.google.android.material.bottomsheet.BottomSheetBehavior;
import com.slimshady.bottomsheetmodule.abstractions.AbstractBottomSheet;

public class BaseBottomSheet<V extends ViewBinding> extends AbstractBottomSheet {

    protected V viewBinding;
    protected Context context;
    protected int parentLayoutBottomSheetResId;

    protected View viewBottomsheet;
    protected BottomSheetBehavior bottomSheetBehavior;
    protected int expandedHeight; //Height of bottom sheet in expanded state
    protected int peekHeight;

    public BaseBottomSheet(V viewBinding, Context context, int parentLayoutBottomSheetResId) {
        this.viewBinding = viewBinding;
        this.context = context;
        this.parentLayoutBottomSheetResId = parentLayoutBottomSheetResId;

        View view = viewBinding.getRoot();
        viewBottomsheet = view.findViewById(parentLayoutBottomSheetResId);
        bottomSheetBehavior = BottomSheetBehavior.from(viewBottomsheet);

    }


    @Override
    protected void initBottomSheetView(Context context) {
        viewBottomsheet.setBackgroundResource(R.drawable.bottom_dialog_shape);
        bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);

        //Retrieve bottom sheet parameters
        ViewGroup.LayoutParams bottomSheetLayoutParams = viewBottomsheet.getLayoutParams();
        bottomSheetLayoutParams.height = getWindowHeight(context);

        expandedHeight = bottomSheetLayoutParams.height;
        peekHeight = (int) (expandedHeight / 2.3); //Peek height to 70% of expanded height (Change based on your view)

        //Setup bottom sheet
        viewBottomsheet.setLayoutParams(bottomSheetLayoutParams);
        BottomSheetBehavior.from(viewBottomsheet).setSkipCollapsed(false);
        BottomSheetBehavior.from(viewBottomsheet).setPeekHeight(peekHeight);
        BottomSheetBehavior.from(viewBottomsheet).setHideable(true);

    }

    @Override
    public void closeBottomSheet() {
        if (bottomSheetBehavior != null) {
            if (bottomSheetBehavior.getState() == BottomSheetBehavior.STATE_EXPANDED ||
                    (bottomSheetBehavior.getState() == BottomSheetBehavior.STATE_COLLAPSED) &&
                            bottomSheetBehavior.getPeekHeight() != 0) {
                bottomSheetBehavior.setPeekHeight(0);
                bottomSheetBehavior.setState(BottomSheetBehavior.STATE_COLLAPSED);

            }
        }
    }

}

File: app/src/main/java/com/slimshady/bottomsheetmodule/BaseBottomSheet.kt
package com.slimshady.bottomsheetmodule

import android.content.Context
import android.view.View
import com.slimshady.bottomsheetmodule.util.ScreenUtils.getWindowHeight
import androidx.viewbinding.ViewBinding
import com.slimshady.bottomsheetmodule.abstractions.AbstractBottomSheet
import com.google.android.material.bottomsheet.BottomSheetBehavior
import com.slimshady.bottomsheetmodule.R
import android.view.ViewGroup

open class BaseBottomSheet<V : ViewBinding>(
    protected var viewBinding: V,
    protected var context: Context,
    protected var parentLayoutBottomSheetResId: Int
) : AbstractBottomSheet() {
    protected var viewBottomsheet: View
    @JvmField
    protected var bottomSheetBehavior: BottomSheetBehavior<*>?
    @JvmField
    protected var expandedHeight //Height of bottom sheet in expanded state
            = 0
    @JvmField
    protected var peekHeight = 0
    override fun initBottomSheetView(context: Context) {
        viewBottomsheet.setBackgroundResource(R.drawable.bottom_dialog_shape)
        bottomSheetBehavior?.state = BottomSheetBehavior.STATE_COLLAPSED

        //Retrieve bottom sheet parameters
        val bottomSheetLayoutParams = viewBottomsheet.layoutParams
        bottomSheetLayoutParams.height = getWindowHeight(context)
        expandedHeight = bottomSheetLayoutParams.height
        peekHeight =
            (expandedHeight / 2.3).toInt() //Peek height to 70% of expanded height (Change based on your view)

        //Setup bottom sheet
        viewBottomsheet.layoutParams = bottomSheetLayoutParams
        BottomSheetBehavior.from(viewBottomsheet).skipCollapsed = false
        BottomSheetBehavior.from(viewBottomsheet).peekHeight = peekHeight
        BottomSheetBehavior.from(viewBottomsheet).isHideable = true
    }

    override fun closeBottomSheet() {
        if (bottomSheetBehavior != null) {
            if (bottomSheetBehavior?.state == BottomSheetBehavior.STATE_EXPANDED ||
                bottomSheetBehavior?.state == BottomSheetBehavior.STATE_COLLAPSED &&
                bottomSheetBehavior?.peekHeight != 0
            ) {
                bottomSheetBehavior?.peekHeight = 0
                bottomSheetBehavior?.state = BottomSheetBehavior.STATE_COLLAPSED
            }
        }
    }

    init {
        val view = viewBinding.root
        viewBottomsheet = view.findViewById(parentLayoutBottomSheetResId)
        bottomSheetBehavior = BottomSheetBehavior.from(viewBottomsheet)
    }
}